Adding objects, controllers, and classes for questions and answers

Alexalex1432 opened the pull request:
Some test util methods were refactored to save as part of providing a test record
Resolving FUN-9413

Commit:
Adding objects, controllers, and classes for questions and answers

Some test util methods were refactored to save as part of providing a test record
Resolving FUN-9413

adelorme82 commented on src/classes/frDonor.cls:
```diff
@@ -102,22 +102,27 @@ public class frDonor extends frModel {
+            Boolean cityAndState = String.isNotBlank(city) && String.isNotBlank(state);
+            if(namePresent && String.isNotBlank(address1) && (cityAndState || String.isNotBlank(postalCode))) {
+                String byAddress = 'select Id, fr_ID__c, LastName from Contact where MailingStreet = :address1' +
+                    ' and MailingCity = :city' +
```
I'm not sure about this thought, but is there anything to only querying on the fields that are in the request? If they include `name AND street address AND cityAndState` but not `postalCode`, should we exclude `postalCode` from the query?

Alexalex1432 replied:
I really like that, I'm going to see if we can incorporate that logic without making the query code too messy

adelorme82 commented on src/classes/frAnswer.cls:
```diff
@@ -0,0 +1,137 @@
+            Supporter__c = supporter != null ? supporter.Id : null,
+            Transaction__c = opp != null ? opp.Id : null,
+            Campaign_Page__c = page != null ? page.Id : null,
+			Registration__c = registration != null ? registration.Id : null
```
The absolute most minor, only fix if there is other stuff we find, there is some whitespace weirdness here

Commit:
Finalizing removal of registration master-detail field in favor of the new lookup field

With this removal, subscriber organizations of the managed package can now remove the master-detail field
Resolving FUN-9413

Commit:
Adding transactionId as a mappable field for opportunity/donation

Resolving FUN-9915

Commit:
Code review feedback - Updating address matching logic to leave off fields that are blank & fixing some whitespace issues

Resolving FUN-9413

Commit:
Including the answer and question tabs on the Funraise app

Resolving FUN-10425
